xarray/core/duck_array_ops.py

```diff
@@ -171,6 +171,83 @@ def _ignore_warnings_if(condition):
         yield
 
 
+def _nansum_object(value, axis=None, **kwargs):
+    """ In house nansum for object array """
+    value = fillna(value, 0)
+    return _dask_or_eager_func('sum')(value, axis=axis, **kwargs)
+
+
+def _nan_minmax_object(func, fill_value_typ, value, axis=None, **kwargs):
+    """ In house nanmin and nanmax for object array """
+    if fill_value_typ == '+inf':
```

Comment:
Nit: Instead of passing a separate string, we might just pass the function to make the fill value directly (`dtypes.get_pos_infinity` or `dtypes.get_neg_infinity`).

That would let us drop these conditionals and error prone string matching.